### DISCUSSION TITLE
Modding: "OnNotContaining" status-effect type.

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#7840] [Ideas and suggestions]
We have "Contained" and "NotContained" status effects for items,
however for the containers themselves we only have "OnContaining" - there is no "OnNotContaining"
IE, the status effect is as "Always" whenever the container is empty.

`(Creating a hacky-workaround 'virtual not-comparison' using other status effects is not always viable if the item already makes use of properties like Condition and IsActive etc.)`

**Comment by mLuby** [comment]
Request for `OnStopUsing` also. A synthetic version works but is probably expensive performance-wise since it uses `Always`:
```xml
<!-- NOTE ignoreContinuousToggle has no actual effect here; using it purely as a flag to indicate OnNotUse. -->
<StatusEffect type="Always" targettype="This" ignoreContinuousToggle="true" setvalue="true" stackable="false" duration="1" />
<StatusEffect type="OnUse" targettype="This" ignoreContinuousToggle="false" setvalue="true" />
```

**Reply by Ek-Videogames** [reply]
Yea, I use a similar constant-on system in my current fix/workaround to ammo loading logic.
